src/ray/gcs/gcs_server/gcs_node_manager.h

```diff
@@ -159,6 +159,9 @@ class GcsNodeManager : public rpc::NodeInfoHandler {
         std::shared_ptr<gcs::GcsPubSub> gcs_pub_sub,
         std::function<void(const ClientID &)> on_node_death_callback);
 
+    /// NOTE: TO AVOID HEARTBEAT BEING DEALYED BY MAIN THREAD, MAKE SURE EVERY PUBLIC
+    /// METHOD BELOW SHOULD BE POSTED TO ITS OWN IO SERVICE.
```

Comment:
```suggestion
```suggestion
    // Note: To avoid heartbeats being delayed by main thread, all public methods below
    // should be posted to its own IO service.
```
```